src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs

```diff
@@ -141,5 +140,102 @@ public void AreRootsEqual(string first, string second, StringComparison comparis
         {
             Assert.Equal(expected, PathInternal.AreRootsEqual(first, second, comparisonType));
         }
+
+        public static TheoryData<string, int> GetRootLength_Data => new TheoryData<string, int>
+        {
+            { @"C:\git\corefx", 3},
+            { @"C:\git\.\", 3},
+            { @"C:\git\..\", 3},
+            { @"C:\git\..\..\", 3},
+            { @"C:\..\", 3},
+            { @"C:\", 3},
+            { @"C:", 2},
+            { @"C:git\corefx", 2},
+            { @"C:git\.\", 2},
+            { @"C:git\..\", 2},
+            { @"C:..\", 2},
+        };
+
+        public static TheoryData<string, int> GetRootLengthRooted => new TheoryData<string, int>
+        {
+            { @"\tmp", 1},
+            { @"\.", 1},
+            { @"\..", 1},
+            { @"\tmp\..\..", 1},
+            { @"\\", 2},
+            { @"\\\", 3},
```

Comment:
"\\\" should be probably be 2, but I'm not too concerned. I only care that initial double separators are considered special and are never coalesced into one separator. Having 3+ in the start is beyond what we want to handle or rationalize.

Reply:
```\\\``` is getting treated as a Unc path here. I am ignoring this input as we are expecting correct separators until volume